ctru-rs/src/services/am.rs

```diff
@@ -0,0 +1,121 @@
+use crate::error::ResultCode;
+use crate::services::fs::FsMediaType;
+use std::marker::PhantomData;
+use std::mem::MaybeUninit;
+
+#[derive(Copy, Clone, Debug)]
+#[repr(C)]
+pub struct TitleInfo {
+    id: u64,
+    size: u64,
+    version: u16,
+    pad: u16,
+    type_: u32,
+}
+
+// Make sure TitleInfo is correct size
+const _TITLEINFO_SIZE_CHECK: [u8; 0x18] = [0; std::mem::size_of::<TitleInfo>()];
```

Comment:
We don't need size checks if we avoid hard-coded allocations. Refer to the comment on `TitleInfo`.

Reply:
Done